src/components/Trade/TradeInput.tsx

```diff
@@ -148,6 +171,16 @@ export const TradeInput = () => {
   const { assetSearch } = useModal()
   const { handleAssetClick } = useTradeRoutes()
 
+  // Trigger re-validation of the manually entered receive address
+  useEffect(() => {
+    formTrigger(SendFormFields.Input)
+  }, [formTrigger])
+
+  // Reset the manual address input state when the user changes the buy asset
+  useEffect(() => {
+    setFormValue(SendFormFields.Input, '')
+  }, [buyAsset, setFormValue])
```

Comment:
`selectBuyAsset` doesnt seem to be a `deepEqualOutputSelector` so for safety should we use the asset id since its primitive?

```suggestion
  }, [buyAsset.assetId, setFormValue])
```

Reply:
probably unnecessary but hey